File: binance4j-nft/src/main/java/com/binance4j/nft/dto/OrderType.java
package com.binance4j.nft.dto;

import io.swagger.annotations.ApiModelProperty;

/** The NFT order type */
public enum OrderType {
	/** Purchase order */
	PURCHASE_ORDER("0"), @ApiModelProperty("")
	/** Sell order */
	SELL_ORDER("1"), @ApiModelProperty("")
	/** Royalty income */
	ROYALTY_INCOME("2"), @ApiModelProperty("")
	/** Primary market_order */
	PRIMARY_MARKET_ORDER("3"), @ApiModelProperty("")
	/** Mint fee */
	MINT_FEE("4");

	/**
	 * @param value The value.
	 */
	private OrderType(String value) {
		this.value = value;
	}

	/** The error code. */
	final String value;

	@Override
	public String toString() {
		return value;
	}
}

File: connectors/src/main/kotlin/com/binance4j/nft/dto/OrderType.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.nft.dto

import com.fasterxml.jackson.annotation.JsonProperty

/** The NFT order type.
 *
 * @property value Value.
 */
enum class OrderType(@JsonProperty("value")
	var value: String) {
	/** Purchase order  */
	PURCHASE_ORDER("0"),
	
	/** Sell order  */
	SELL_ORDER("1"),
	
	/** Royalty income  */
	ROYALTY_INCOME("2"),
	
	/** Primary market_order  */
	PRIMARY_MARKET_ORDER("3"),
	
	/** Mint fee  */
	MINT_FEE("4");
	
	override fun toString(): String = value
}